### PR TITLE
Add trafficlight feedback

### PR DESCRIPTION
**Pull request by MaHaWo**
**backend**
- add  function for computing milestonegroup feedback: 
- baseline:
  - get all answers to group's milestones relevant for a requested age range 
  - compute avg, standard deviation from them. this is the baseline we compare against

- feedback:
  - too kinds: detailed (list of milestone feedbacks) and summarized (see below):
  - for the summarized feedback per milestonegroup:
  - get answer session for a certain child of a certain user
  - get the milestonegroups in the answersession 
  - for each milestonegroup: 
       - calculate statistics baseline as discussed above 
       - get mean of the answers of the current milestonegroup --> **this is maybe not a good choice, what did they want/ how should results of milestonegroups be aggregated and evaluated? The average can hide individual delayed milestones in a large collection of normally developing milestones**
       - compare the mean result of the current child (=m) over the current milestonegroup with the population avg: 
           - m smaller than population avg - 2 sigma: -1, negative, should do something 
           - avg - 2 * sigma < m < avg-sigma:  0, not so bad, but look out for further development
           - m > avg - sigma: +1,  alright, no need to do anything 
- add routers for summarized and individual detailed feedback
- add missing tests
- **frontend** 
  - have  a timeline of answer sessions (currently has no cutoff, so all are displayed) 
  - show the worst milestonegroup result of the answersession as the overall assessment of it to prompt investigation by user 
  - have an accordion for the milestonegroups with its result displayed. 
  - click on accordion item gives detailed feedback for each assessed milestone and shows "how to give aid" text via a popover upon click 
  - should bad results in milestones bubble up more?

### CODE REVIEW

File: mondey_backend/src/mondey_backend/routers/utils.py [hunk 1]
```diff
@@ -182,47 +183,125 @@ def _get_expected_age_from_scores(scores: np.ndarray) -> int:
...
+    milestonegroup = get(session, MilestoneGroup, mid)
+    answers = []
+    for milestone in milestonegroup.milestones:
+        # we want something that is relevant for the age of the child at hand. Hence we filter by age here. Is this what they want?
+        m_answers = [
+            answer.answer
+            for answer in session.exec(
+                select(MilestoneAnswer).where(
+                    col(MilestoneAnswer.milestone_id) == milestone.id
+                    and age_lower <= milestone.expected_age_months <= age_upper
```
Comment (lkeegan) [inline mondey_backend/src/mondey_backend/routers/utils.py:285]: see other comment about using .where().where() or `&` instead

**Review by lkeegan — APPROVED** [review]
backend part looks good - some comments/suggestions.

also I think it would be good for you to talk me through the logic, I got a bit lost in the details :-)

File: mondey_backend/src/mondey_backend/routers/utils.py [hunk 2]
```diff
@@ -119,7 +120,7 @@ def update_child_question_text(session: SessionDep, child_question: ChildQuestio
 
 
 def _session_has_expired(milestone_answer_session: MilestoneAnswerSession) -> bool:
-    session_lifetime_days = 7
+    session_lifetime_days = 3
```
[the remark this answers is not in the document]
Comment (MaHaWo) [reply mondey_backend/src/mondey_backend/routers/utils.py:123]: this is a temporary change. shouldn´t be merged. thx.

File: mondey_backend/src/mondey_backend/models/milestones.py [hunk 3]
```diff
@@ -174,11 +174,34 @@ class MilestoneAnswerSessionPublic(SQLModel):
...
     scores: list[MilestoneAgeScore]
     expected_age: int
+
+
+class MilestoneScore(SQLModel, table=True):
+    child_id: int = Field(default=None, foreign_key="child.id", primary_key=True)
+    milestone_id: int = Field(
+        default=None, foreign_key="milestone.id", primary_key=True
+    )
+    score: float
```
[the remark this answers is not in the document]
Comment (MaHaWo) [reply mondey_backend/src/mondey_backend/models/milestones.py:194]: yes, in `MilestoneScore**s**.` Has been for a while, so I didn´t dare change it.

File: mondey_backend/src/mondey_backend/routers/utils.py [hunk 4]
```diff
@@ -182,47 +183,125 @@ def _get_expected_age_from_scores(scores: np.ndarray) -> int:
...
             MilestoneAgeScore(
+                milestone_id=milestone_id,
                 age_months=age,
-                avg_score=score,
-                expected_score=(4 if age >= expected_age else 1),
+                avg_score=avg[age],
+                stddev_score=stddev[age],
+                expected_score=(
+                    4 if age >= expected_age else 1
+                ),  # FIXME: don´t know what this is supposed to mean
```
[the remark this answers is not in the document]
Comment (MaHaWo) [reply mondey_backend/src/mondey_backend/routers/utils.py:258]: thanks!

File: mondey_backend/src/mondey_backend/routers/users.py [hunk 5]
```diff
@@ -235,4 +243,92 @@ def update_current_child_answers(
...
+    )
+    def get_expired_milestone_answer_sessions(
+        session: SessionDep, current_active_user: CurrentActiveUserDep, child_id: int
+    ) -> dict[int, MilestoneAnswerSessionPublic]:
+        milestone_answer_sessions = {
+            mas.id: mas  # type: ignore
+            for mas in session.exec(
+                select(MilestoneAnswerSession).where(
+                    col(MilestoneAnswerSession.user_id) == current_active_user.id
+                    and col(MilestoneAnswerSession.child_id) == child_id
```
[the remark this answers is not in the document]
Comment (MaHaWo) [reply mondey_backend/src/mondey_backend/routers/users.py:258]: true. thanks. Ran into this issue at another point, but this one escaped me.

**Pull request closed without merging**